dumbo/src/tcp/connection.rs

```diff
@@ -466,9 +465,12 @@ impl Connection {
         }
 
         if v + current_w > max_w {
```

Comment:
I know this was like that before, but since you're modifying this method, can you also change this line to `if v > max_w - current_w {`? It makes it more resilient to issues caused by supplying very large values for `v`.